controller.go

```diff
@@ -63,17 +72,18 @@ func (c *K8SController) mainloop(item interface{}) {
 		c.OnUpdate(obj)
 	case ControllerEventTypeDelete:
 		c.OnDelete(obj)
+	default:
+		log.Warn().Msgf("no event handler for '%s', event type '%s'", event.Key, event.Type)
 	}
-	c.queue.Done(item)
 }
 
-// Start - starts the informer faktory and sync's the data.
+// Start - starts the informer factory and syncs the data.
 // The wait group passed in is used to track when the informer has gone
-// through 1 full loop and syncronized all the k8s data 1 time
+// through 1 full loop and syncronized all the k8s data exactly 1 time
 func (c *K8SController) Start(wg *sync.WaitGroup) {
 	defer runtime.HandleCrash()
-	defer c.queue.ShutDown()
 	if wg != nil {
+		defer c.queue.ShutDown()
```

Comment:
why does this need to move into this logic block.  This will still need to be called if the waitgroup is nil and we are only running one mainloop.

Reply:
Closing this since we discussed this will be addressed in https://github.com/OpsLevel/team-platform/issues/184 (it's not as simple as just moving the shutdown call into the goroutine, we need a context or channel to do this.)